### PR TITLE
Add a round timer

### LINKED ISSUE #74: add a clock/round timer for tournaments.

### PR DESCRIPTION
**Pull request by haradan**
Resolves #74

### CODE REVIEW

File: app/services/round_timer.rb [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+
+  def reset!
+    round_timer_activations.destroy_all
+  end
+
+  def show?
+    round_timer_activations.count > 0 && !completed?
+  end
+
+  def paused?
```
Comment (plural) [inline app/services/round_timer.rb:23]: Can the States you've defined below be used here instead of the poking around on last?

File: spec/services/nrtm_json_spec.rb [hunk 2]
```diff
@@ -54,11 +54,11 @@
       let(:json) { described_class.new(tournament) }
 
       before do
-        r1 = create(:round, stage: cut, completed: true)
+        r1 = create(:round, number: 2, stage: cut, completed: true)
```
Comment (plural) [inline spec/services/nrtm_json_spec.rb:57]: why are these changes included?

**Review by plural — COMMENTED** [review]
looks pretty good!  Just a couple of things from me.

File: app/services/round_timer.rb (hunk 1 above)
Comment (haradan) [reply app/services/round_timer.rb:23]: I'm not sure if there's a good way to do it. The state function needs to check if it's paused first before it knows what else needs to be calculated. In order to know if it's paused it needs to add up all the previous activations. It's not a lot of computation since there won't be many activations, but it would be nice to feel like it ties together a bit better.

File: spec/services/nrtm_json_spec.rb (hunk 2 above)
Comment (haradan) [reply spec/services/nrtm_json_spec.rb:57]: I had a problem where this test failed locally because the rounds showed up in the wrong order. The model for round has `default_scope { order(number: :asc) }`, so I think setting the round number should solve the problem. It always gets set in real usage.

Comment (plural) [reply spec/services/nrtm_json_spec.rb:57]: ah, nice!  I had seen some non-determinism in tests but not that one yet.

**Review by plural — APPROVED** [review]
